ament_cmake_core/ament_cmake_uninstall_target-extras.cmake

```diff
@@ -32,6 +32,8 @@ if(AMENT_CMAKE_UNINSTALL_TARGET)
   )
 
   # register uninstall target to run generated CMake script
-  add_custom_target(uninstall
-    COMMAND ${CMAKE_COMMAND} -P "${AMENT_CMAKE_UNINSTALL_TARGET_UNINSTALL_SCRIPT}")
+  if(NOT TARGET uninstall)
```

Comment:
If the target exists (as in your cases when building multiple packages in a single CMake context) just skipping this implies that the uninstall logic is not triggered for the additional packages. I don't think this is sufficient.